src/pudl/transform/nrelatb.py

```diff
@@ -0,0 +1,300 @@
+
+    rate_table: TableUnstacker
+    scenario_table: TableUnstacker
+    tech_detail_table: TableUnstacker
+
+    @property
+    def params_all(self: Self) -> list[str]:
+        """Compilation of all of the parameter values from each of the tables.
+
+        Also check if there are no duplicate params. We expect all of the parameters
+        across the :class:`TableUnstacker` to be unique.
+        """
+        params_all = (
+            self.rate_table.params
+            + self.scenario_table.params
+            + self.tech_detail_table.params
+        )
+        assert not {x for x in params_all if params_all.count(x) > 1}
+        return params_all
+
+
+def unstacker_of_tables() -> Unstacker:
+    """Build an instance of :class:`Unstacker` for the core NREL ATB tables.
+
+    This function is soley for easy access to the reshaping parameters.
+    """
+    unstacker_rate_table = TableUnstacker(
+        idx=[
+            "report_year",
+            "mystery_code",
+            "core_metric_case",
+            "core_metric_variable_year",
+            "cost_recovery_period_years",
+            "technology_description",
+            "core_metric_parameter",
+        ],
+        params=[
+            "inflation rate",
+            "interest during construction - nominal",
+            "tax rate (federal and state)",
+        ],
+    )
+    unstacker_scenario_table = TableUnstacker(
+        idx=unstacker_rate_table.idx + ["scenario_atb"],
+        params=[
+            "interest rate nominal",
+            "debt fraction",
+            "wacc real",
+            "wacc nominal",
+            "calculated interest rate real",
+            "rate of return on equity nominal",
+            "calculated rate of return on equity real",
+            # 2023 only param
+            "crf",
+        ],
+    )
+    unstacker_detail_table = TableUnstacker(
+        idx=unstacker_scenario_table.idx
+        + [
+            "technology_description_detail_1",
+            "technology_description_detail_2",
+        ],
+        params=[
+            "capex",
+            "cf",
+            "fixed o&m",
+            "fuel",
+            "lcoe",
+            "variable o&m",
+            "fcr",
+            # 2023 only params
+            "additional occ",
+            "cfc",
+            "gcc",
+            "heat rate",
+            "heat rate penalty",
+            "net output penalty",
+            "occ",
+        ],
+    )
+    unstack_tables = Unstacker(
+        rate_table=unstacker_rate_table,
+        scenario_table=unstacker_scenario_table,
+        tech_detail_table=unstacker_detail_table,
+    )
+    return unstack_tables
+
+
+@asset
+def _core_nrelatb__transform_start(raw_nrelatb__data):
+    """Transform raw NREL ATB data into semi."""
+    ## EARLY CLEANING/DEDUPING
+    raw_nrelatb__data["mystery_code"] = raw_nrelatb__data.core_metric_key.str[0]
```

Comment:
`mystery_code`... still is a mystery to me. we need to figure out what this code means. it is either `M` or `R` but that does not seem to correspond to Market/R&D in the `core_metric_case` column.

Reply:
i made this moot by removed two of the oldest years of data #3576